source/patterns/@aws-solutions-constructs/aws-eventbridge-sqs/lib/index.ts

```diff
@@ -151,18 +175,27 @@ export class EventbridgeToSqs extends Construct {
     this.encryptionKey = buildQueueResponse.key;
     this.deadLetterQueue = buildQueueResponse.dlq;
 
-    let sqsEventTargetProps: eventtargets.SqsQueueProps = {};
+    let constructEventTargetProps: eventtargets.SqsQueueProps = {};
 
-    if (defaults.CheckBooleanWithDefault(props.deployRuleDlq, false)) {
-      this.eventRuleDlq = defaults.buildQueue(this, 'ruleDlq', {
+    if (defaults.CheckBooleanWithDefault(props.deployEventRuleDlq, false)) {
+
+      const buildRuleDlqResponse = defaults.buildQueue(this, 'ruleDlq', {
         deployDeadLetterQueue: false,
         enableEncryptionWithCustomerManagedKey: enableEncryptionParam,
-        encryptionKey: this.encryptionKey,
-      }).queue;
+        encryptionKeyProps: props.eventRuleDlqKeyProps
+      });
+
+      this.eventRuleDlq = buildRuleDlqResponse.queue;
+      const ruleDlqKey = buildRuleDlqResponse.key;
+      ruleDlqKey?.grantEncryptDecrypt(new ServicePrincipal('events.amazonaws.com'));
+      this.eventRuleDlqKey = ruleDlqKey;
+
+      // TODO: Update docs on encrpytWithCustomerMasterKey
```

Comment:
Just checking, did we want to keep this TODO in here for future or were we supposed to action on this as part of this PR?